若不传递Key参数会导致错误

```
traefik-reverse-proxy-1  | time="2023-11-23T13:49:01Z" level=error msg="plugins-local/src/github.com/SwanHubX/req-preprocess/preprocess.go:127:14: panic" plugin=plugin-hello module=github.com/SwanHubX/req-preprocess
traefik-reverse-proxy-1  | time="2023-11-23T13:49:01Z" level=error msg="plugins-local/src/github.com/SwanHubX/req-preprocess/preprocess.go:48:29: panic" plugin=plugin-hello module=github.com/SwanHubX/req-preprocess
traefik-reverse-proxy-1  | time="2023-11-23T13:49:01Z" level=error msg="Error in Go routine: reflect: call of reflect.Value.Field on zero Value"
traefik-reverse-proxy-1  | time="2023-11-23T13:49:01Z" level=error msg="Stack: goroutine 76 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:24 +0x5e\ngithub.com/traefik/traefik/v2/pkg/safe.defaultRecoverGoroutine({0x4288980?, 0xc001219e78})\n\tgithub.com/traefik/traefik/v2/pkg/safe/routine.go:66 +0x9e\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1.1()\n\tgithub.com/traefik/traefik/v2/pkg/safe/routine.go:56 +0x34\npanic({0x4288980?, 0xc001219e78?})\n\truntime/panic.go:914 +0x21f\ngithub.com/traefik/yaegi/interp.runCfg.func1()\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:205 +0x185\npanic({0x4288980?, 0xc001219e78?})\n\truntime/panic.go:914 +0x21f\ngithub.com/traefik/yaegi/interp.runCfg.func1()\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:205 +0x185\npanic({0x4288980?, 0xc001219e78?})\n\truntime/panic.go:914 +0x21f\nreflect.Value.Field({0x0?, 0x0?, 0x3eef040?}, 0x0?)\n\treflect/value.go:1278 +0xcf\nreflect.Value.FieldByIndex({0x0?, 0x0?, 0xc0007e0000?}, {0xc000fa2db8?, 0xc000fa7a10?, 0xc000e01fb0?})\n\treflect/value.go:1311 +0x4e\ngithub.com/traefik/yaegi/interp.getIndexSeqField.func3(0xc000cffad0?)\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:2137 +0xca\ngithub.com/traefik/yaegi/interp.runCfg(0xc000f857c0, 0xc000cffad0, 0x5670?, 0xc001240378?)\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:213 +0x285\ngithub.com/traefik/yaegi/interp.call.func9(0xc000cffa20)\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:1389 +0x7a9\ngithub.com/traefik/yaegi/interp.runCfg(0xc000f6a500, 0xc000cffa20, 0xc?, 0x4049a20?)\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:213 +0x285\ngithub.com/traefik/yaegi/interp.genFunctionWrapper.func1.1({0xc000ca98c0, 0x4, 0x8?})\n\tgithub.com/traefik/yaegi@v0.15.1/interp/run.go:1024 +0x7c5\nreflect.Value.call({0xc0009b4a80?, 0xc000ffc8d0?, 0x7fe489b9a868?}, {0x4f10de4, 0x4}, {0xc0012413d8, 0x4, 0x0?})\n\treflect/value.go:596 +0xce7\nreflect.Value.Call({0xc0009b4a80?, 0xc000ffc8d0?, 0x7fe489b91f18?}, {0xc0012413d8?, 0xc000500000?, 0xc000e01dd0?})\n\treflect/value.go:380 +0xb9\ngithub.com/traefik/traefik/v2/pkg/plugins.middlewareBuilder.newHandler({{0xc0009b4a80, 0xc000ffc8d0, 0x13}, {0xc000073780, 0xc000ffcba0, 0x13}}, {0x5c72820?, 0xc000e01da0?}, {0x5c18b40, 0xc000e01d70}, ...)\n\tgithub.com/traefik/traefik/v2/pkg/plugins/middlewares.go:62 +0x296\ngithub.com/traefik/traefik/v2/pkg/plugins.(*Middleware).NewHandler(...)\n\tgithub.com/traefik/traefik/v2/pkg/plugins/middlewares.go:132\ngithub.com/traefik/traefik/v2/pkg/server/middleware.newTraceablePlugin({0x5c72820?, 0xc000e01da0?}, {0xc000d1cde0, 0xf}, 0xc000e01e00?, {0x5c18b40?, 0xc000e01d70?})\n\tgithub.com/traefik/traefik/v2/pkg/server/middleware/plugins.go:45 +0x3f\ngithub.com/traefik/traefik/v2/pkg/server/middleware.(*Builder).buildConstructor.func23({0x5c18b40?, 0xc000e01d70?})\n\tgithub.com/traefik/traefik/v2/pkg/server/middleware/middlewares.go:360 +0x3d\ngithub.com/traefik/traefik/v2/pkg/server/middleware.(*Builder).buildConstructor.Wrap.func24({0x5c18b40?, 0xc000e01d70?})\n\tgithub.com/traefik/traefik/v2/pkg/middlewares/tracing/wrapper.go:24 +0x51\ngithub.com/traefik/traefik/v2/pkg/server/middleware.(*Builder).BuildChain.func1({0x5c18b40, 0xc000e01d70})\n\tgithub.com/traefik/traefik/v2/pkg/server/middleware/middlewares.go:81 +0xee\ngithub.com/containous/alice.Chain.Then({{0xc00007a8b0?, 0x5c72820?, 0xc000e01aa0?}}, {0x5c174a0?, 0xc00007a860?})\n\tgithub.com/containous/alice@v0.0.0-20181107144136-d83ebdd94cbd/chain.go:51 +0x89\ngithub.com/traefik/traefik/v2/pkg/server/router.(*Manager).buildHTTPHandler(0xc001241dd8, {0x5c72820?, 0xc000e01aa0}, 0xc00064a9b0, {0xc000ffa9f0, 0xe})\n\tgithub.com/traefik/traefik/v2/pkg/server/router/router.go:205 +0x76c\ngithub.com/traefik/traefik/v2/pkg/server/router.(*Manager).buildRouterHandler(0xc001241dd8, {0x5c72820, 0xc000e01aa0}, {0xc000ffa9f0, 0xe}, 0xc00064a9b0)\n\tgithub.com/traefik/traefik/v2/pkg/server/router/router.go:159 +0x18d\ngithub.com/traefik/traefik/v2/pkg/server/router.(*Manager).buildEntryPointHandler(0x5c727b0?, {0x5c72820?, 0xc000e01980}, 0x1?)\n\tgithub.com/traefik/traefik/v2/pkg/server/router/router.go:118 +0x1d8\ngithub.com/traefik/traefik/v2/pkg/server/router.(*Manager).BuildHandlers(0xc001241dd8, {0x5c727b0, 0x8c27680}, {0xc0007e0640?, 0x2, 0x2}, 0x0?)\n\tgithub.com/traefik/traefik/v2/pkg/server/router/router.go:72 +0x24f\ngithub.com/traefik/traefik/v2/pkg/server.(*RouterFactory).CreateRouters(0xc0009b4b60, 0xc000c9ca00)\n\tgithub.com/traefik/traefik/v2/pkg/server/routerfactory.go:77 +0x20e\nmain.setupServer.switchRouter.func6({0xc000d91f50?, 0xc0012187e0?, 0xc00120b790?, 0xc000d91f80?})\n\tgithub.com/traefik/traefik/v2/cmd/traefik/traefik.go:360 +0x52\ngithub.com/traefik/traefik/v2/pkg/server.(*ConfigurationWatcher).applyConfigurations(0xc0009b4bd0, {0x5c72858, 0xc0008b5220})\n\tgithub.com/traefik/traefik/v2/pkg/server/configurationwatcher.go:172 +0x1fc\ngithub.com/traefik/traefik/v2/pkg/safe.(*Pool).GoCtx.func1()\n\tgithub.com/traefik/traefik/v2/pkg/safe/routine.go:36 +0x57\ngithub.com/traefik/traefik/v2/pkg/safe.GoWithRecover.func1()\n\tgithub.com/traefik/traefik/v2/pkg/safe/routine.go:59 +0x4d\ncreated by github.com/traefik/traefik/v2/pkg/safe.GoWithRecover in goroutine 1\n\tgithub.com/traefik/traefik/v2/pkg/safe/routine.go:53 +0x67\n"
```
大多数时候我们仅使用一种方式认证，因此可能并没有另一个参数

--- a/preprocess.go
+++ b/preprocess.go
@@ -45,7 +45,10 @@ type JWT struct {
 }
 
 func New(ctx context.Context, next http.Handler, config *Config, name string) (http.Handler, error) {
-	publicKey := loadPublicKey(config.Key) // 加载pem公钥
+	var publicKey *rsa.PublicKey = nil
+	if config.Key != "" {
+		publicKey = loadPublicKey(config.Key) // 加载pem公钥
+	}
 	return &Preprocess{
 		next:      next,
 		name:      name,

